### PR TITLE
docs(extensions): document the new before/after extension points

### LINKED ISSUE #244: document the new before/after extension capabilities
there is no documentation for https://github.com/adobe/helix-pipeline/pull/211

### PR DESCRIPTION
**Pull request by trieloff**
fixes #244

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +148,62 @@ If in the above example, the `doSomething` causes an error, subsequently, `rende
...
+  .once(render)
+  .after(cleanup).expose('cleanup')
+  .after(done);
+```
+
+In this example, two extension points, `init` and `cleanup` have been defined. Note how the name of the extension point can be the same as the name of the function (i.e. `cleanup`), but does not have to be the same (i.e. `init` vs. `doSomething`).
+
+#### Common Extension Points
+
+The creation of extension points are the responsibility of the pipeline author, but in order to standardize extension points, following common names have been established:
```
Comment (rofe) [inline README.md]: Grammar: 
- The creation (...) **is** the responsibility of (...)
- (...) **the** following common names (...)

File: README.md [hunk 2]
```diff
@@ -148,6 +148,62 @@ If in the above example, the `doSomething` causes an error, subsequently, `rende
...
+}
+
+module.exports.after = {
+  fetch: (context, action) => {
+    // will get called after the "fetch" pipeline step
+  }
+}
+```
+
+All functions that are using the `before` and `after` extension points need to folllow the same interface that all other pipeline functions follow, i.e. they have access to `context` and `action` and they should return a modified `context` object.
```
Comment (rofe) [inline README.md]: Typo: **follow**

File: README.md (hunk 1 above)
Comment (trieloff) [reply README.md]:
```suggestion
The creation of extension points is the responsibility of the pipeline author, but in order to standardize extension points, the following common names have been established:
```

File: README.md (hunk 2 above)
Comment (trieloff) [reply README.md]:
```suggestion
All functions that are using the `before` and `after` extension points need to follow the same interface that all other pipeline functions follow, i.e. they have access to `context` and `action` and they should return a modified `context` object.
```

**Pull request merged**